utils.js

```diff
@@ -16,15 +16,37 @@ const Display = global.display;
 export let version = Config.PACKAGE_VERSION.split('.').map(Number);
 
 let warpRipple;
+
+let touchSignal = null;
+let inTouch = false;
+let touchCoords = null;
+
 export function enable() {
     warpRipple = new Ripples.Ripples(0.5, 0.5, 'ripple-pointer-location');
     warpRipple.addTo(Main.uiGroup);
+
+    touchSignal = global.stage.connect("captured-event", (actor, event) => {
+        switch (event.type()) {
+        case Clutter.EventType.TOUCH_BEGIN:
+        case Clutter.EventType.TOUCH_UPDATE:
+            inTouch = true;
+            break;
+        case Clutter.EventType.TOUCH_END:
+        case Clutter.EventType.TOUCH_CANCEL:
+            inTouch = false;
+            break;
+        }
+        touchCoords = event.get_coords();
```

Comment:
This will be called for other events (not just touch events) which is likely not the intent here.  Add a `default` case that simply calls `return Clutter.EVENT_PROPAGATE;`

The `captured-event` is rather heavy, and captures all events (clicks, swipes, activations, etc.) so it's important that it's as efficient as possible, and filtering on touch events etc.